lib/iris/fileformats/netcdf.py

```diff
@@ -1865,47 +1893,32 @@ def _create_cf_data_variable(self, cube, dimension_names, local_keys=None,
             The newly created CF-netCDF data variable.
 
         """
-        if 'packing' in kwargs:
-            packing = kwargs.pop('packing')
-            if packing:
-                scale_factor = None
-                add_offset = None
-                fill_value = None
-                if isinstance(packing, dict):
-                    if 'dtype' not in packing:
-                        msg = "The dtype attribute is required for packing."
-                        raise ValueError(msg)
-                    dtype = np.dtype(packing['dtype'])
-                    if 'scale_factor' in packing:
-                        scale_factor = packing['scale_factor']
-                    if 'add_offset' in packing:
-                        add_offset = packing['add_offset']
-                    if 'fill_value' in packing:
-                        fill_value = packing['fill_value']
+
+        if packing:
+            if isinstance(packing, dict):
+                if 'dtype' not in packing:
+                    msg = "The dtype attribute is required for packing."
+                    raise ValueError(msg)
+                dtype = np.dtype(packing['dtype'])
+                scale_factor = packing.get('scale_factor', None)
+                add_offset = packing.get('add_offset', None)
```

Comment:
Not your code, but would you mind checking that there aren't other keys in there other than the ones handled?